### ISSUE TITLE
Bugs encountered while looking at examples

### ISSUE THREAD

**Issue by AccelQuasarDragon** [mp-007/kivy_matplotlib_widget#23]
https://youtu.be/qGE8CFq3xuI

bug 1 + fix: 
In example_4_differents_subplot, join should be joined, you can see it at 31:07

bug 2:
In several examples, (like exampe_minmax_option), there is an issue with cursor on windows. I always get a "y must be a sequence". You can see it at 28:30

bug 3: 
Sometimes zoom does not work, see 20:33

**Comment by mp-007** [comment]
Thanks for all these reported bug. I will work on it when I have time.

For the first bug, I will used "sharey" function to fix this (was add in matplotlib v3.3). My code was using a old code from stackoverflow,

**Comment by mp-007** [comment]
@AccelQuasarDragon 

For bug 3. it's not a bug. 
The reason why your zoombox it's not working, it's because your zoombox selection it's lower then your "minzoom' attribute. By default this parameter is dp(40). You can change this attribute the a minimum value of dp(20) if you wanted. Plotly package use something similar to avoid zooming small selection.

**Comment by mp-007** [comment]
@AccelQuasarDragon 

for bug2
Can you give be your numpy version and matplotlib version?
I'm not able to reproduce this error with matplotlib 3.5.2 and numpy 1.21.5.
To have this error, you need to set a not iterable y data input when you create your figure.

**Comment by mp-007** [comment]
Update:

matplotlib cursor with register_lines function doesn't work with matplotlib 3.9. Need to fix xdata, ydata format when used matplotlib cursor.

**Comment by mp-007** [comment]
@AccelQuasarDragon 

bug 1 and 2 should be fix in version 0.11.1 (if you still using matplotlib 3.9). Please feel free to close this issue if your are happy with all my changes and answer.

**Comment by AccelQuasarDragon** [comment]
Sorry, I did not see the notification on github. Thanks for pinging me on discord. 

TODO: 
example_mplcursors still crashes with `raise RuntimeError('y must be a sequence')`
I checked on both a poetry add virtualenv and a pip install virtualenv. I think this is the last thing to update.

I also checked these examples as of kivy-matplotlib-widget 0.12.4, they work now!
example_minmax_option WORKS
example_4_differents_subplot WORKS
example_cursor WORKS
example_hover WORKS

also, if you see a red dot or red circle when you right click, that is not your fault, it's kivy's multitouch emulation as per: https://stackoverflow.com/a/37572966/16355112

![rightclick](https://github.com/mp-007/kivy_matplotlib_widget/assets/138998466/9d311792-52be-4de7-a947-0086b357fa33)

matplotlib, numpy version for the video: 
kivy_matplotlib_widget 0.11.0
matplotlib             3.9.0
numpy                  2.0.0


I made another repo to get the latest matplotlib widget, it most of the bugged examples now work fine with this setup: 
kivy-matplotlib-widget 0.12.4
matplotlib             3.9.1
numpy                  2.0.0

**Comment by mp-007** [comment]
> Sorry, I did not see the notification on github. Thanks for pinging me on discord.
> 
> TODO: example_mplcursors still crashes with `raise RuntimeError('y must be a sequence')` I checked on both a poetry add virtualenv and a pip install virtualenv. I think this is the last thing to update.
> 
> I also checked these examples as of kivy-matplotlib-widget 0.12.4, they work now! example_minmax_option WORKS example_4_differents_subplot WORKS example_cursor WORKS example_hover WORKS
> 
> also, if you see a red dot or red circle when you right click, that is not your fault, it's kivy's multitouch emulation as per: https://stackoverflow.com/a/37572966/16355112
> 
> ![rightclick](https://private-user-images.githubusercontent.com/138998466/347022480-9d311792-52be-4de7-a947-0086b357fa33.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjA1NDAxNjMsIm5iZiI6MTcyMDUzOTg2MywicGF0aCI6Ii8xMzg5OTg0NjYvMzQ3MDIyNDgwLTlkMzExNzkyLTUyYmUtNGRlNy1hOTQ3LTAwODZiMzU3ZmEzMy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzA5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcwOVQxNTQ0MjNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03ZjU3NmM4Yzg5ZTkyZTZjMmJlYzQwNTI2OWNkZTNkN2IxOTAwNmQzODM3N2ZjM2Q5M2VmOWExM2UwZDQ4YjFiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.2AU4Jf4rWizWKQ4eiv7EY1a8xb_rjSIWE6PqIfRY9tY)
> 
> matplotlib, numpy version for the video: kivy_matplotlib_widget 0.11.0 matplotlib 3.9.0 numpy 2.0.0
> 
> I made another repo to get the latest matplotlib widget, it most of the bugged examples now work fine with this setup: kivy-matplotlib-widget 0.12.4 matplotlib 3.9.1 numpy 2.0.0

Did a fix on commit f3388864cc8c55f95ee0e53964677c65300ee549 for mpl_cursors example
and remove red dot on commit 2107ccb3e227681d05136e30e114efafb748b0fd for the 3D example

**Comment by AccelQuasarDragon** [comment]
what I did:
looked at the github changelog, and all the changes were on the widget examples themselves
therefore, I deleted old cloned repo, cloned the newer repo with the same virtualenv
cd into tests and checked them out

example_mplcursors WORKS

re: the red dots
the 3d example is fixed. 
I think all the other examples have red dots since it's a Kivy's base multitouch feature. Just letting people know it's a thing. In my opinion not a big deal as the fix is easily googleable and not the focus of the matplotlib project. 

Thanks for the quick fixes and nice plots/examples!  Closing with this comment.
